bull/pdl-queue/consumer.js

```diff
@@ -66,7 +68,40 @@ async function getZipAndBytelength(no_of_pages, id, title, job) {
   return [zip, byteLength, errorFlag];
 }
 
-function setHeaders(metadata, byteLength, title) {
+async function getPdfAndBytelength(pdfUrl, job) {
+  try {
+    const response = await customFetch(
+      pdfUrl,
+      "GET",
+      new Headers({
+        "Content-Type": "application/pdf",
+      }),
+      "file"
+    );
+    if (response.status === 200) {
+      job.progress(30);
+      const buffer = await response.buffer();
+      job.progress(60);
+      return {
+        pdfBuffer: buffer,
+        byteLength: buffer.byteLength,
+      };
+    } else {
+      logger.log({
+        level: "error",
+        message: `Failure PDL: Failed to download PDF. Status Code: ${response.status}`,
+      });
+    }
+  } catch (error) {
+    logger.log({
+      level: "error",
+      message: `Failure PDL: ${error}`,
+    });
+    return null;
+  }
+}
+
+function setHeaders(metadata, byteLength, title, contentType) {
```

Comment:
@coderwassananmol 
forgot to use it inside the function, have corrected it.
Its been added to the metadata  headers["Content-type"] = `application/${contentType}`;
Since the content might be either application/zip or application/pdf